src/state/slices/preferencesSlice/preferencesSlice.ts

```diff
@@ -176,5 +179,14 @@ export const preferences = createSlice({
     setSnapInstalled(state, { payload }: { payload: boolean }) {
       state.snapInstalled = payload
     },
+    addWatchedAsset(state, { payload }: { payload: AssetId }) {
+      state.watchedAssets = [...state.watchedAssets, payload]
```

Comment:
*If* we have guarantees from the consumer/s (which seems to be the case) this is never going to be called with an asset that's already present in the watched assets list, then we could simply
```suggestion
      state.watchedAssets = state.watchedAssets.concat(payload)
```
(assuming `payload` is now an array, else replace with `[payload]`)

Most likely not a big deal perf-wise unless we have a huge list, but might as well